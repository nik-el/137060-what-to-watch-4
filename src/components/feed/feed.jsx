import React from 'react';
import PropTypes from 'prop-types';
import {Card} from '../card/card';
import {FilmPropTypes} from "../../types/film-prop-types";

export class Feed extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      activeCard: null
    };

    this.handleCardHover = (index) => {
      this.setState({
        activeCard: index
      });
    };
  }

  render() {
    const {films, onCardTitleClick, className} = this.props;

    return (
      <div className={className}>
        { films.map((item) =>
          (
            <Card
              title={item.title}
              thumbnail={item.thumbnail}
              key={item.id}
              onCardMouseOver={() => this.handleCardHover(item)}
              onCardTitleClick={onCardTitleClick}
            />
          )
        )}
      </div>
    );
  }
}

Feed.propTypes = {
  // массив данных с фильмами
  films: PropTypes.arrayOf(PropTypes.shape(FilmPropTypes)),
  // обработчик клика по заголовку карточки
  onCardTitleClick: PropTypes.func.isRequired,
  className: PropTypes.string
};
