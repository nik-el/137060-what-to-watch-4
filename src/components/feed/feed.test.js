import React from 'react';
import renderer from 'react-test-renderer';
import {Feed} from './feed';

const onCardTitleClick = () => {};

const testFilms = [
  {
    title: `Test 1`,
    thumbnail: `https://picsum.photos/id/1`,
    id: `1`,
  },
  {
    title: `Test 2`,
    thumbnail: `https://picsum.photos/id/2`,
    id: `2`,
  },
];

it(`Feed renders correctly`, () => {
  const tree = renderer
    .create(<Feed
      films={testFilms}
      onCardTitleClick={onCardTitleClick}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

