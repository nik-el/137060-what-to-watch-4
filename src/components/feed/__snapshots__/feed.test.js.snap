// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<div>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Test 1"
        height="175"
        src="https://picsum.photos/id/1"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Test 1
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Test 2"
        height="175"
        src="https://picsum.photos/id/2"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Test 2
      </a>
    </h3>
  </article>
</div>
`;
